fplll/nr/nr_Z.inl

```diff
@@ -208,22 +208,39 @@ public:
   inline void swap(Z_NR<Z>& a);
 
   /**
-   * Generates random integer between 0 and 2^bits-1. 
+   * Generates random integer between 0 and 2^bits-1.
    */
   inline void randb(int bits);
   inline void randb_si(int bits);
 
   /**
-   * Generates random integer between 0 and max - 1  
+   * Generates random integer between 0 and max - 1
    */
   inline void randm(const Z_NR<Z>& max);
   inline void randm_si(const Z_NR<Z>& max);
 
-  /**	 
-   * Generates smallest prime number above nbr 
+  /**
+   * Generates smallest prime number above nbr
   */
   inline void nextprime(const Z_NR<Z>& nbr);
 
+  /**
```

Comment:
I don't like this strategy: it leads to "flexible" templated code which isn't flexible at all. LinBox was guilty of that for a long time: you'd have to guess/know which templates can actually be instantiated. I'd rather have these for all data types unless there is a very good reason.

Reply:
For now I've limited myself to mpz_t for HNF computations. The main function should return an error when tried to force a non-gmp integer type for HNF computations. I can try to code those number theoretical function for all data types, it will take some extra time. To be fair I will probably add some for modular computations.

I added those in the integer class for the sole purpose to make the HNF code more readible, which isn't templated (yet?).